cmd/cephcsi.go

```diff
@@ -0,0 +1,113 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"flag"
+	"os"
+
+	"github.com/ceph/ceph-csi/pkg/cephfs"
+	"github.com/ceph/ceph-csi/pkg/rbd"
+	"github.com/ceph/ceph-csi/pkg/util"
+	"k8s.io/klog"
+)
+
+const (
+	rbdType    = "rbd"
+	cephfsType = "cephfs"
+
+	rbdDefaultName    = "rbd.csi.ceph.com"
+	cephfsDefaultName = "cephfs.csi.ceph.com"
+)
+
+var (
+	// common flags
+	vtype           = flag.String("type", "", "driver type [rbd|cephfs]")
+	endpoint        = flag.String("endpoint", "unix://tmp/csi.sock", "CSI endpoint")
+	driverName      = flag.String("drivername", "", "name of the driver")
+	nodeID          = flag.String("nodeid", "", "node id")
+	metadataStorage = flag.String("metadatastorage", "", "metadata persistence method [node|k8s_configmap]")
+
+	// rbd related flags
+	containerized = flag.Bool("containerized", true, "whether run as containerized")
+
+	// cephfs related flags
+	volumeMounter = flag.String("volumemounter", "", "default volume mounter (possible options are 'kernel', 'fuse')")
+)
+
+func init() {
+	klog.InitFlags(nil)
+	if err := flag.Set("logtostderr", "true"); err != nil {
+		klog.Exitf("failed to set logtostderr flag: %v", err)
+	}
+	flag.Parse()
+}
+
+func getDriverName() string {
+	// was explicitly passed a driver name
+	if *driverName != "" {
+		return *driverName
+	}
+	// select driver name based on volume type
+	switch *vtype {
+	case rbdType:
+		return rbdDefaultName
+	case cephfsType:
+		return cephfsDefaultName
+	default:
+		return ""
+	}
+}
+
+func main() {
+	if *vtype == "" {
+		klog.Fatalln("driver type not specified")
+	}
+
+	dname := getDriverName()
+	err := util.ValidateDriverName(dname)
+	if err != nil {
+		klog.Fatalln(err) // calls exit
+	}
+
+	switch *vtype {
```

Comment:
`switch dnam` makes more sense based on the context

Reply:
dname can be user specified as well, hence switching on the same may not be the right thing to do.